core/renderers/measurables/rows.js

```diff
@@ -481,3 +481,12 @@ Blockly.blockRendering.InputRow.prototype.measure = function() {
   }
   this.widthWithConnectedBlocks = this.width + connectedBlockWidths;
 };
+
+
+Blockly.blockRendering.InputRow.prototype.getLastSpacer = function() {
+  if (this.hasExternalInput || this.hasStatement) {
+    var spacer = this.elements[this.elements.length - 3];
```

Comment:
This seems very fragile.

Reply:
I made it very very very slightly less fragile.